tsl/src/chunk_api.c

```diff
@@ -371,3 +391,129 @@ chunk_create(PG_FUNCTION_ARGS)
 
 	PG_RETURN_DATUM(HeapTupleGetDatum(tuple));
 }
+
+/*
+ * Detach a chunk from a hypertable.
+ */
+Datum
+chunk_detach(PG_FUNCTION_ARGS)
+{
+	Oid chunk_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	Cache *hcache;
+	Hypertable *ht;
+	Chunk *chunk;
+
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+
+	chunk = ts_chunk_get_by_relid(chunk_relid, true);
+	Assert(chunk != NULL);
+
+	/* Take the same locks taken by PostgreSQL partitioning to be consistent */
+	LockRelationOid(chunk->hypertable_relid, ShareUpdateExclusiveLock);
+	LockRelationOid(chunk->table_id, AccessExclusiveLock);
+
+	/* Ensure that chunk is still there after taking locks */
+	chunk = ts_chunk_get_by_relid(chunk_relid, true);
+	Assert(chunk != NULL);
```

Comment:
I don't think you need this check since you would not be able to lock the chunk using the table_id if it went away.